apps/aevm/src/aevm_ae_primops.erl

```diff
@@ -666,6 +675,13 @@ crypto_call(_Gas, ?PRIM_CALL_CRYPTO_BLAKE2B_STRING, Data, State) ->
 crypto_call(_, _, _, _) ->
     {error, out_of_gas}.
 
+crypto_call_ecrecover_secp256k1(_Gas, Data, State) ->
+    [MsgHash0, Sig0] = get_args([hash_t(), bytes_t(65)], Data),
+    MsgHash = <<MsgHash0:32/unit:8>>,
+    <<_:31/binary, Sig:65/binary>> = words_to_bin(96, Sig0),
```

Comment:
This is wrong. Byte arrays are padded on the right, not on the left. See

https://github.com/aeternity/aebytecode/blob/befa1e3ff9c53c1054917c82e3b30c1c3c743f4f/src/aeb_memory.erl#L13-L18

Suggests that there also need to be tests that goes through the proper calldata creation instead of the hack in `aecontract_SUITE`.

Reply:
Yes, it should be `Sig = words_to_bin(65, Sig0)`...

Reply:
Thanks. Fixed.